docs/troubleshooting-guide/datawarehouse.md

```diff
@@ -1,82 +1,65 @@
+## Troubleshooting connections to the database server
 
-**How to detect this:** If you've verified that you can connect to the data warehouse's host and port, the next step is to check your credentials. Again, connecting to a data warehouse depends on your database server software; for PostgreSQL, a command like the one shown below will do the job:
+1. [Check that the data warehouse server is running](../administration-guide/01-managing-databases.html#checking-the-server-status).
 
-```
-psql -h HOSTNAME -p PORT -d DATABASENAME -U DATABASEUSER`
-```
+2. Check if you can connect to the data warehouse from another client using the machine that you’re running Metabase on.
 
-If your credentials are incorrect, you should see an error message letting you know if the database name or the user/password are incorrect.
+    > If you can access the server from a bastion host or another machine, [check if your Metabase's IP address has access to your database server](../administration-guide/01-managing-databases.html#checking-your-server-access).
 
-**How to fix this:** If the database name or the user/password combination are incorrect, ask the person running your data warehouse for correct credentials.
+    > If you're running Metabase Cloud, check that you've [whitelisted our IP addresses](/cloud/docs/ip-addresses-to-whitelist.html).
 
-<h2 id="connection-timeout-took-too-long">Connection timeout: your question took too long</h2>
+The steps above will help you detect whether the problem is occurring outside of Metabase. To _fix_ problems with your database server, you'll need to refer to the docs for your database or cloud service.
 
-**How to detect this:** If you see the error message, "Your question took too long," something in your setup timed out. Depending on the specifics of your deployment, the problem could be in:
+If you don't have access to the data warehouse server, you’ll need to ask the person who manages your database or data warehouse.
 
-- your load balancer;
-- your reverse proxy server (e.g., Nginx);
-- Jetty;
-- your database; or
-- your cloud service, such as AWS's Elastic Beanstalk, EC2, Heroku, or Google App Engine.
+## Common database connection errors
 
-**How to fix this:** Fixing this depends on your specific setup. These resources may help:
+**From the Metabase interface:**
+- [Your question took too long](./timeout.html).
 
-- [Configuring Jetty connectors][configuring-jetty]
-- [EC2 Troubleshooting][ec2-troubleshooting]
-- [Elastic Load Balancing Connection Timeout Management][elb-timeout]
-- [Heroku timeouts][heroku-timeout]
-- [App Engine: Dealing with DeadlineExceededErrors][app-engine-timeout]
+**From the logs:**
+- [Connections cannot be acquired from the underlying database](#connections-cannot-be-acquired-from-the-underlying-database).
 
-<h2 id="connections-cannot-be-acquired">Connections cannot be acquired from the underlying database</h2>
+### Connections cannot be acquired from the underlying database
```

Comment:
This section needs restructuring. You're linking to a heading that's the very next line.